### PR TITLE
Adds more share actions when browsing a repository (#2161)

### PR DESCRIPTION
**Pull request by ashleyng**
Adds in the feature for more share options when browsing a repository (#2161)
- Adds the `...` when viewing a directory
- Changes the share button to `...` when viewing a file
- Adds "Copy Path", "Copy Name", "Share URL", and "Share Content" options
- each option will open the share sheet giving you the option to share with a different app or just copy it to the clipboard.

For a file `Classes/Issues/Files/IssueFileCell.swift`
- Copy path: `Classes/Issues/Files/IssueFileCell.swift`
- Copy name: `IssueFileCell.swift`
- Share URL: `https://github.com/GitHawkApp/GitHawk/blob/master/Classes/Issues/Files/IssueFileCell.swift`

For a directory `Classes/Issues/Files`
- Copy path: `Classes/Issues/Files`
- Copy Name: `Files`
- Share URL: `https://github.com/GitHawkApp/GitHawk/tree/master/Classes/Issues/Files`

simulator screen shot - iphone 8 - 2018-10-06 at 12 00 54 simulator screen shot - iphone 8 - 2018-10-06 at 12 00 58

### CODE REVIEW

**Comment by rnystrom** [comment]
This is so good! Thank you!!

File: Classes/Utility/AlertAction.swift [hunk 1]
```diff
@@ -21,6 +21,24 @@ struct AlertAction {
...
+        case shareFileName
+        case `default`
+
+        var localizedString: String {
+            switch self {
+            case .shareUrl: return NSLocalizedString("Share URL", comment: "")
+            case .shareContent: return NSLocalizedString("Share Content", comment: "")
+            case .shareFilePath: return NSLocalizedString("Copy Path", comment: "")
+            case .shareFileName: return NSLocalizedString("Copy Name", comment: "")
+            default: return NSLocalizedString("Share", comment: "")
```
Comment (BasThomas) [inline Classes/Utility/AlertAction.swift]: should this be `case .default` instead?

**Comment by BasThomas** [comment]
Awesome stuff @ashleyng!

File: Classes/Utility/AlertAction.swift (hunk 1 above)
Comment (ashleyng) [reply Classes/Utility/AlertAction.swift]: I figured it would be a good fallback if there were any other new sharing options to be added.

Comment (ashleyng) [reply Classes/Utility/AlertAction.swift]: That makes sense. I’ll be making another PR soon for #1958 and can add the change for this in there

**Pull request merged**